tests/src/test/scala/com/nvidia/spark/rapids/RapidsHostMemoryStoreSuite.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.io.File
+
+import ai.rapids.cudf.{ContiguousTable, HostMemoryBuffer, Table}
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.{spy, verify}
+import org.scalatest.FunSuite
+import org.scalatest.mockito.MockitoSugar
+
+import org.apache.spark.sql.rapids.RapidsDiskBlockManager
+
+class RapidsHostMemoryStoreSuite extends FunSuite with Arm with MockitoSugar {
+  private def buildContiguousTable(): ContiguousTable = {
+    withResource(new Table.TestBuilder()
+        .column(5, null.asInstanceOf[java.lang.Integer], 3, 1)
+        .column("five", "two", null, null)
+        .column(5.0, 2.0, 3.0, 1.0)
+        .build()) { table =>
+      table.contiguousSplit()(0)
+    }
+  }
+
+  test("spill updates catalog") {
+    val bufferId = MockRapidsBufferId(7)
+    val spillPriority = -7
+    val hostStoreMaxSize = 1L * 1024 * 1024
+    val catalog = spy(new RapidsBufferCatalog)
+    withResource(new RapidsDeviceMemoryStore(catalog)) { devStore =>
+      withResource(new RapidsHostMemoryStore(catalog, hostStoreMaxSize)) { hostStore =>
+        assertResult(0)(hostStore.currentSize)
+        assertResult(hostStoreMaxSize)(hostStore.numBytesFree)
+        devStore.setSpillStore(hostStore)
+
+        val ct = buildContiguousTable()
+        val bufferSize = ct.getBuffer.getLength
+        try {
+          devStore.addTable(bufferId, ct.getTable, ct.getBuffer, spillPriority)
+        } catch {
+          case t: Throwable =>
+            ct.close()
```

Comment:
nit: Could we add a comment about ownership of the table is transferred to `devStore`. It just didn't fit with a lot of the other patterns we have and It threw me for a bit.

Reply:
`addTable` does have scaladoc explaining the ownership situation, but I noticed it was missing from `addBuffer` for some reason.  I added it there and also added it to the places it was called in the test suites to make it clear.